android/java/org/chromium/chrome/browser/crypto_wallet/presenters/Amount.java

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.presenters;
+
+import android.text.TextUtils;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+public class Amount {
+    private BigDecimal mValue;
+
+    public Amount(BigDecimal value) {
+        if (value == null) {
+            this.mValue = BigDecimal.ZERO;
+            return;
+        }
+        this.mValue = value;
+    }
+
+    public Amount(String value) {
+        if (TextUtils.isEmpty(value)) {
+            this.mValue = BigDecimal.ZERO;
+            return;
+        }
+        this.mValue = new BigDecimal(value);
+    }
+
+    public Amount(double value) {
+        this.mValue = new BigDecimal(value);
+    }
+
+    public Amount add(BigDecimal value) {
+        BigDecimal safeVal = safeVal(value);
+        mValue = mValue.add(safeVal);
+        return this;
+    }
+
+    public Amount add(Amount value) {
+        BigDecimal safeVal = safeVal(value);
+        mValue = mValue.add(safeVal);
+        return this;
+    }
+
+    // Default max decimal precision is 8
+    public String toStringFormat() {
+        return toStringFormat(8);
+    }
+
+    public String toStringFormat(int decimalLength) {
+        return mValue.setScale(decimalLength, RoundingMode.HALF_UP)
+                .stripTrailingZeros()
+                .toPlainString();
+    }
+
+    public String toHex() {
+        if (mValue.signum() == 0) return "0x0";
+        return "0x" + mValue.toBigInteger().toString(16);
+    }
+
+    private BigDecimal safeVal(BigDecimal value) {
+        safeInit();
+        return (value == null) ? BigDecimal.ZERO : value;
+    }
+
+    private BigDecimal safeVal(Amount value) {
+        safeInit();
```

Comment:
same here as above, how `Amount value` and that whole class are related? Why do we need to call some `safeInit` of that class if we don't use `mValue` from it at all?

Reply:
here it is not required. will remove.

Reply:
same as above I don't get why it's not static. Any class methods have to manipulate with class members or access to them for a read only purpose. But a class function that makes a decision based on a passed value shouldn't have an access to the class if it doesn't need it.

Reply:
Same as [above](https://github.com/brave/brave-core/pull/15593#discussion_r1007970736).